### PR TITLE
update reference/features/vindex

### PR DESCRIPTION
**Pull request by sougou**
Signed-off-by: Sugu Sougoumarane <ssougou@gmail.com>

### CODE REVIEW

File: content/en/docs/reference/features/vindexes.md [hunk 1]
```diff
@@ -110,6 +110,27 @@ Of the above categories, `Functional Unique` and `Lookup Unique Unowned` Vindexe
...
+        "from": "name",
+        "to": "keyspace_id"
+      },
+      "owner": "user"
+    }
+```
+
+In the above case, the name of the vindex is `name_keyspace_idx`. It is of type `lookup`, and it is owned by the `user` table.
+
+Every Vindex has an optional `params` section that contains a map of string key-value pairs. The keys and values defer depending on the vindex type and are explained below.
```
Comment (deepthi) [inline content/en/docs/reference/features/vindexes.md]:
```suggestion
Every Vindex has an optional `params` section that contains a map of string key-value pairs. The keys and values differ depending on the vindex type and are explained below.
```

File: content/en/docs/reference/features/vindexes.md [hunk 2]
```diff
@@ -147,27 +168,42 @@ If the table owns lookup vindexes, then the rows to be deleted are first read an
...
-region\_json | Functional Unique | Multi-column prefix-based hash combined with a JSON map for key-to-region mapping, for use in geo-partitioning | Yes | No | 1 | String and numeric type |
-reverse\_bits | Functional Unique | Bit reversal | Yes | Yes | 1 | 64 bit or smaller numeric or equivalent type |
-unicode\_loose\_md5 | Functional Unique | Case-insensitive (UCA level 1) MD5 hash | Yes | No | 1 | String or binary types |
-unicode\_loose\_xxhash | Functional Unique | Case-insensitive (UCA level 1) xxHash64 hash | Yes | No | 1 | String or binary types |
-xxhash | Functional Unique | xxHash64 hash | Yes | No | 1 | Any |
+Name | Type | Description | Primary | Multi-column | Reversible | Cost | Data types |
+---- | ---- | ----------- | ------- | ------------ | ---------- | ---- | ---------- |
+binary | Functional Unique | Identity | Yes | No | Yes | 0 | Any |
+binary\_md5 | Functional Unique | MD5 hash | Yes | No | No | 1 | Any |
+consistent\_lookup | Lookup NonUnique | Lookup table non-unique values | No | Identify Row | No | 20 | Any |
```
Comment (deepthi) [inline content/en/docs/reference/features/vindexes.md]: What does it mean that "Multi-column" = "Identify Row"?

File: content/en/docs/reference/features/vindexes.md [hunk 3]
```diff
@@ -147,27 +168,42 @@ If the table owns lookup vindexes, then the rows to be deleted are first read an
...
+consistent\_lookup\_unique | Lookup Unique | Lookup table unique values | If unowned | Identify Row | No | 10 | Any |
+hash | Functional Unique | DES null-key hash | Yes | No | Yes | 1 | 64 bit or smaller numeric or equivalent type |
+lookup | Lookup NonUnique | Lookup table non-unique values | No | Identify Row | No | 20 | Any |
+lookup\_unique | Lookup Unique | Lookup table unique values | If unowned | Identify Row | No | 10 | Any |
+null | Functional Unique | Always map to keyspace ID 0 | Yes | No | No | 100 | Any |
+numeric | Functional Unique | Identity | Yes | No | Yes | 0 | 64 bit or smaller numeric or equivalent type |
+numeric\_static\_map | Functional Unique | JSON file statically mapping input string values to keyspace IDs | Yes | No | No | 1 | Any |
+region\_experimental | Functional Unique | Multi-column prefix-based hash for use in geo-partitioning | Yes | No | No | 1 | String and numeric type |
+region\_json | Functional Unique | Multi-column prefix-based hash combined with a JSON map for key-to-region mapping, for use in geo-partitioning | Yes | No | No | 1 | String and numeric type |
```
Comment (deepthi) [inline content/en/docs/reference/features/vindexes.md]: Multi-column Yes

File: content/en/docs/reference/features/vindexes.md [hunk 4]
```diff
@@ -147,27 +168,42 @@ If the table owns lookup vindexes, then the rows to be deleted are first read an
...
+binary\_md5 | Functional Unique | MD5 hash | Yes | No | No | 1 | Any |
+consistent\_lookup | Lookup NonUnique | Lookup table non-unique values | No | Identify Row | No | 20 | Any |
+consistent\_lookup\_unique | Lookup Unique | Lookup table unique values | If unowned | Identify Row | No | 10 | Any |
+hash | Functional Unique | DES null-key hash | Yes | No | Yes | 1 | 64 bit or smaller numeric or equivalent type |
+lookup | Lookup NonUnique | Lookup table non-unique values | No | Identify Row | No | 20 | Any |
+lookup\_unique | Lookup Unique | Lookup table unique values | If unowned | Identify Row | No | 10 | Any |
+null | Functional Unique | Always map to keyspace ID 0 | Yes | No | No | 100 | Any |
+numeric | Functional Unique | Identity | Yes | No | Yes | 0 | 64 bit or smaller numeric or equivalent type |
+numeric\_static\_map | Functional Unique | JSON file statically mapping input string values to keyspace IDs | Yes | No | No | 1 | Any |
+region\_experimental | Functional Unique | Multi-column prefix-based hash for use in geo-partitioning | Yes | No | No | 1 | String and numeric type |
```
Comment (deepthi) [inline content/en/docs/reference/features/vindexes.md]: Multi-column Yes

**Comment by rohit-nayak-ps** [comment]
@sougou, looks like this still have pending review comments

File: content/en/docs/reference/features/vindexes.md [hunk 5]
```diff
@@ -147,27 +168,42 @@ If the table owns lookup vindexes, then the rows to be deleted are first read an
...
 
 Consistent lookup vindexes, as described above, are a new category of Vindexes that are meant to replace the existing lookup Vindexes implementation. For the time being, they have a different name to allow for users to switch back and forth.
 
+Lookup Vindexes support the following parameters:
+
+* `table`: The backing table for the lookup vindex. It is recommended that the table name be qualified by its keyspace.
+* `from`: The list of "from" columns. The first column is used for routing, and the rest of the columns are used for identifying the owner row.
+* `to`: The name of the "to" keyspace\_id column.
+* `autocommit` (false): if true, vindex entries are updated in their own autocommit transaction. This is useful if values never get remapped to different values. For example, if the input column comes from an auto-increment value.
+* `write_only` (false): if true, the vindex is kept updated, but a lookup will return all shards if the key is not found. This mode is used while the vindex is being populated and backfilled.
```
Comment (jmoldow) [inline content/en/docs/reference/features/vindexes.md:211]: > but a lookup will return all shards if the key is not found

This is not true, as discussed in https://github.com/vitessio/vitess/issues/7336 and on Slack.

Comment (deepthi) [reply content/en/docs/reference/features/vindexes.md:211]: I suggest adding this to the documentation as a known issue, and we can remove that once it is fixed.

**Review by harshit-gangal — COMMENTED** [review]
LGTM once the comments are addressed.

**Comment by rohit-nayak-ps** [comment]
@sougou bump

File: content/en/docs/reference/features/vindexes.md (hunk 2 above)
Comment (sougou) [reply content/en/docs/reference/features/vindexes.md]: Added an explanation.

File: content/en/docs/reference/features/vindexes.md (hunk 5 above)
Comment (deepthi) [reply content/en/docs/reference/features/vindexes.md:211]: Having read the linked issue, I don't think we need to call that out in this document. Let us merge this PR as is.

**Review by deepthi — APPROVED** [review]

**Pull request merged**